### PR TITLE
✨([dogwood|eucalyptus]/3/[fun|wb]) force libcast as default video xblock

### PR DESCRIPTION
**Pull request by rmoch**
This forces edx to use `libcast_xblock` as default video xblock
instead of youtube by providing a function that will replace `video`
entry point by `libcast _xblock`'s one each time an xblock is instanciated.

### CODE REVIEW

File: releases/dogwood/3/fun/config/lms/fun.py [hunk 1]
```diff
@@ -379,3 +379,27 @@ def __iter__(self):
 ANALYTICS_DASHBOARD_URL = config(
     "ANALYTICS_DASHBOARD_URL", default=False, formatter=bool
 )
+
+# This force Edx Studio to use `libcast_xblock` as default video player
+# in the studio (big green button) and if an xblock is called `video`
+FUN_DEFAULT_VIDEO_PLAYER = 'libcast_xblock'
+
+def prefer_fun_xmodules(identifier, entry_points):
```
Comment (sampaccoud) [inline releases/dogwood/3/fun/config/lms/fun.py]: maybe put this function in the utils.py file so it is shared with the cms?

Comment (rmoch) [reply releases/dogwood/3/fun/config/lms/fun.py]: 👍

File: releases/eucalyptus/3/wb/config/lms/utils.py [hunk 2]
```diff
@@ -108,3 +108,25 @@ def get(self, name, *args, **kwargs):
...
+    if identifier == "video":
+        import pkg_resources
+        from xblock.core import XBlock
+        # These entry points are listed in the setup.py of the libcast module
+        # Inspired by the XBlock.load_class method
+        entry_points = list(
+            pkg_resources.iter_entry_points(
+                XBlock.entry_point,
+                name="libcast_xblock"
+                )
```
Comment (sampaccoud) [inline releases/eucalyptus/3/wb/config/lms/utils.py]: some wrong identation for these parentheses

File: releases/dogwood/3/fun/config/lms/utils.py [hunk 3]
```diff
@@ -107,3 +107,25 @@ def get(self, name, *args, **kwargs):
...
+    if identifier == "video":
+        import pkg_resources
+        from xblock.core import XBlock
+        # These entry points are listed in the setup.py of the libcast module
+        # Inspired by the XBlock.load_class method
+        entry_points = list(
+            pkg_resources.iter_entry_points(
+                XBlock.entry_point,
+                name="libcast_xblock"
+                )
```
Comment (sampaccoud) [inline releases/dogwood/3/fun/config/lms/utils.py]: some wrong identation for these parentheses

File: releases/dogwood/3/fun/config/lms/utils.py [hunk 4]
```diff
@@ -107,3 +107,25 @@ def get(self, name, *args, **kwargs):
             #    - make a PR to Open edX to provide a better default for this setting.
             default = None
         return self(name, default=default)
+
+
+def prefer_fun_video(identifier, entry_points):
+    """
+    This function will be affected to XBLOCK_SELECT_FUNCTION which used to filter
```
Comment (sampaccoud) [inline releases/dogwood/3/fun/config/lms/utils.py]:
```suggestion
    This function will be affected to XBLOCK_SELECT_FUNCTION which is used to filter
```

File: releases/eucalyptus/3/wb/config/lms/utils.py [hunk 5]
```diff
@@ -108,3 +108,25 @@ def get(self, name, *args, **kwargs):
             #    - make a PR to Open edX to provide a better default for this setting.
             default = None
         return self(name, default=default)
+
+
+def prefer_fun_video(identifier, entry_points):
+    """
+    This function will be affected to XBLOCK_SELECT_FUNCTION which used to filter
```
Comment (sampaccoud) [inline releases/eucalyptus/3/wb/config/lms/utils.py]:
```suggestion
    This function will be affected to XBLOCK_SELECT_FUNCTION which is used to filter
```

**Comment by rmoch** [comment]
`make demo-course` do not work with this modification and it makes the ci fail

File: releases/dogwood/3/fun/CHANGELOG.md [hunk 6]
```diff
@@ -9,6 +9,11 @@ release.
 
 ## [Unreleased]
 
+### Added
+
+- Force edX to use `libcast_xblock` as default video xblock
+- Specific demo course for FUN's flavors
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/CHANGELOG.md]:
```suggestion
- Use custom demo course for FUN's flavors
```

File: releases/dogwood/3/fun/activate [hunk 7]
```diff
@@ -3,4 +3,4 @@ export FLAVOR="fun"
 export EDX_RELEASE_REF="fun-5.2.0"
 export EDX_ARCHIVE_URL="https://github.com/openfun/edx-platform/archive/${EDX_RELEASE_REF}.tar.gz"
 export EDX_DEMO_RELEASE_REF="open-release/eucalyptus.1"
-export EDX_DEMO_ARCHIVE_URL="https://github.com/edx/edx-demo-course/archive/${EDX_DEMO_RELEASE_REF}.tar.gz"
+export EDX_DEMO_ARCHIVE_URL="file://${PWD}/releases/FUN-demo-course-dogwood-eucalyptus.tar.gz"
```
Comment (jmaupetit) [inline releases/dogwood/3/fun/activate]: A carriage return is missing.

File: releases/eucalyptus/3/wb/CHANGELOG.md [hunk 8]
```diff
@@ -9,6 +9,11 @@ release.
 
 ## [Unreleased]
 
+### Added
+
+- Force edX to use `libcast_xblock` as default video xblock
+- Specific demo course for FUN's flavors
```
Comment (jmaupetit) [inline releases/eucalyptus/3/wb/CHANGELOG.md]:
```suggestion
- Use custom demo course for FUN's flavors
```

File: releases/dogwood/3/fun/activate (hunk 7 above)
Comment (jmaupetit) [inline releases/dogwood/3/fun/activate]: I think you should move (and eventually copy for eucalyptus/3/wb) the demo course to the flavor's root:
```suggestion
export EDX_DEMO_ARCHIVE_URL="file://${PWD}/releases/dogwood/3/fun/demo-course.tar.gz"
```

File: releases/eucalyptus/3/wb/activate [hunk 9]
```diff
@@ -3,3 +3,4 @@ export FLAVOR="wb"
 export EDX_RELEASE_REF="fun/whitebrand"
 export EDX_ARCHIVE_URL="https://github.com/openfun/edx-platform/archive/${EDX_RELEASE_REF}.tar.gz"
 export EDX_DEMO_RELEASE_REF="open-release/eucalyptus.3"
+export EDX_DEMO_ARCHIVE_URL="file://${PWD}/releases/FUN-demo-course-dogwood-eucalyptus.tar.gz"
```
Comment (jmaupetit) [inline releases/eucalyptus/3/wb/activate]:
```suggestion
export EDX_DEMO_ARCHIVE_URL="file://${PWD}/releases/eucalyptus/3/wb/demo-course.tar.gz"
```

Comment (jmaupetit) [inline releases/eucalyptus/3/wb/activate]: Don't forget the carriage return!

**Comment by rmoch** [comment]
Ok, so it did not existed yet in dogwood. no reference in edx source code.

File: releases/dogwood/3/fun/activate (hunk 7 above)
Comment (rmoch) [reply releases/dogwood/3/fun/activate]: ok

**Pull request merged**